agent/consul/auto_config_endpoint.go

```diff
@@ -84,87 +79,86 @@ func (a *jwtAuthorizer) Authorize(req *agentpb.AutoConfigRequest) (AutoConfigOpt
 	}, nil
 }
 
-// Cluster endpoint is used for cluster configuration operations
-type Cluster struct {
-	srv *Server
+type AutoConfigBackend interface {
+	GetConfig() *Config
+	CreateACLToken(template *structs.ACLToken) (*structs.ACLToken, error)
+	DatacenterJoinAddresses(segment string) ([]string, error)
+	TLSConfigurator() *tlsutil.Configurator
+	ForwardRPC(method string, info structs.RPCInfo, args, reply interface{}) (bool, error)
+}
 
+// AutoConfig endpoint is used for cluster auto configuration operations
+type AutoConfig struct {
+	backend    AutoConfigBackend
 	authorizer AutoConfigAuthorizer
 }
 
+// getBackendConfig is a small wrapper around the backend's GetConfig to ensure
+// that all helper functions have a non-nil configuration to operate on.
```

Comment:
I think your comment convinced me to do the extra work to allow passing in the configuration so it will be a moot point.

I could go either way on where to default the config. On the one hand adding implied but compiler unenforceable constraints on the return value doesn't sit great with me. Then again the usage of calling the private method as an intermediary to fixup the potential nil everywhere isn't great either.